server/src/test/java/org/opensearch/action/RenamedTimeoutRequestParameterTests.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action;
+
+import org.junit.AfterClass;
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.rest.action.cat.RestNodesAction;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.rest.FakeRestRequest;
+import org.opensearch.threadpool.TestThreadPool;
+
+import static org.hamcrest.Matchers.containsString;
+
+/**
+ * As of 2.0, the request parameter 'master_timeout' in all applicable REST APIs is deprecated,
+ * and alternative parameter 'cluster_manager_timeout' is added.
+ * The tests are used to validate the behavior about the renamed request parameter.
+ * Remove the test after removing MASTER_ROLE and 'master_timeout'.
+ */
+public class RenamedTimeoutRequestParameterTests extends OpenSearchTestCase {
+    private static final TestThreadPool threadPool = new TestThreadPool(RenamedTimeoutRequestParameterTests.class.getName());
+    private final NodeClient client = new NodeClient(Settings.EMPTY, threadPool);
+
+    private static final String DUPLICATE_PARAMETER_ERROR_MESSAGE =
+        "Please only use one of the request parameters [master_timeout, cluster_manager_timeout].";
+    private static final String MASTER_TIMEOUT_DEPRECATED_MESSAGE =
+        "Deprecated parameter [master_timeout] used. To promote inclusive language, please use [cluster_manager_timeout] instead. It will be unsupported in a future major version.";
+
+    @AfterClass
+    public static void terminateThreadPool() {
+        terminate(threadPool);
+    }
+
+    /**
+     * Validate both parameters 'cluster_manager_timeout' and its predecessor can be parsed correctly.
+     */
+    public void testCatAllocation() {
```

Comment:
You should probably break this into 3 separate test methods, something like:
- testNoWarningsForNewParam
- testDeprecationWarningForOldParam
- testBothParamsNotValid

Then you don't need the comments because the test methods are self-explanatory, and if there was any failure it would point to a more specific cause.

Reply:
> You never know what kind of test the next developer will add to this test class :)

Haha, looks like I should explicitly state do not adding new tests to this class, I'm joking. Though it is a temporary and tedious test case.

Reply:
Well, the method `testCatAllocation()` is not a unit test for the method `RenamedTimeoutRequestParameterTests()`, but testing the CAT Nodes API can parse the 2 parameters properly. The problem is that there are 60 other REST APIs to be adding new request parameter `cluster_manager_timeout`, if I break that method to 3 separate ones, it will result 180 new method to be added 🤯.
Instead, I think I'd better add unit tests for the method `RenamedTimeoutRequestParameterTests()`, and create 3 test method as you suggested. And maybe skip unit test for specific REST APIs.

Reply:
I'm just suggesting that you break this test method down into 3 separate tests instead of 1, since it is actually testing 3 separate things. I think it's fine to test this behavior in this one class and was not suggesting that you add tests to all other APIs.

Reply:
I see 😄. I agree to test the behavior in this class, and I will modify the unit tests.

Reply:
Thanks for your suggestions! I modified to test one behavior in a single test case. For the other REST APIs that has got "master_timeout" parameter, I will add one test case for each.